crypto4pl.c

```diff
@@ -66,6 +68,255 @@ typedef enum
 } crypt_mode_t;
 
 
+                 /***************************
+                 *         HASHING          *
+                 ****************************/
+
+
+#define CONTEXT_MAGIC (~ 0x53481284L)
+
+typedef struct context
+{ int             magic;
+  atom_t          atom;
```

Comment:
Ok.  This was Matt's work.  I normally put a little comment with struct members and typically use `symbol` for this one.  Don't bother.